src/components/Form/Form.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import usersFromServer from '../../api/users';
+import { Todo } from '../../types/Todo';
+import { getUserById } from '../../services/user';
+import todosFromServer from '../../api/todos';
+
+type Props = {
+  onSubmit: (todo: Todo) => void;
+};
+
+export const Form: React.FC<Props> = ({ onSubmit }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+  const completed = false;
+
+  const maxId = Math.max(...todosFromServer.map(todo => todo.id), 0);
+
+  function handleTitleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  }
+
+  function handleUserId(event: React.ChangeEvent<HTMLSelectElement>) {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  }
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onSubmit({
+      title,
+      id: maxId + 1,
```

Comment:
The `id` calculation logic should be moved to the `App` component. The `Form` component should receive the `id` as part of the new todo object from the `App` component.